src/sortStudents.ts

```diff
@@ -1,16 +1,72 @@
+/* eslint-disable max-len */
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(students: Student[], sortBy: SortType, order: SortOrder): Student[] {
+  const newArr: Student[] = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  switch (sortBy) {
+    case SortType.Name:
+      if (order === 'asc') {
+        return newArr.sort((a, b) => a.name.localeCompare(b.name));
+      }
+
+      return newArr.sort((a, b) => b.name.localeCompare(a.name));
+
+    case SortType.Surname:
+      if (order === 'asc') {
+        return newArr.sort((a, b) => a.surname.localeCompare(b.surname));
+      }
+
+      return newArr.sort((a, b) => b.surname.localeCompare(a.surname));
```

Comment:
```suggestion
    case SortType.Name:
    case SortType.Surname:
      if (order === 'asc') {
        return newArr.sort((a, b) => a[sortBy].localeCompare(b[sortBy));
      }

      return newArr.sort((a, b) => b[sortBy].localeCompare(a[sortBy]));
```
Можна замість явних 'name' чи 'surname' використовувати змінну sortBy, але використати квадратні дужки. Тоді можна об'єднати ці два кейси.

Також замість if для return'а можна використати тернарник